services/beacon.rb

```diff
@@ -0,0 +1,47 @@
+# encoding: utf-8
+
+module AppOptics::Services
+  class Service::Beacon < AppOptics::Services::Service
+    ENDPOINT = ENV['BEACON_API_ENDPOINT']
+
+    def receive_validate(errors)
+      [:api_key, :description].each do |k|
+        errors[k] = "is required" if settings[k].to_s.empty?
+      end
+      errors.none?
+    end
+
+    def receive_alert_clear
+    end
+
+    def receive_alert
+      raise_config_error unless receive_validate({})
+
+      body = {
+        alert_definition_id: "appoptics-#{payload[:alert][:id]}",
+        # alert_instance_id: "#{payload[:alert][:name]} + #{tags...}"
+        alert_instance_id: payload[:incident_key],
+        alert_instance_origination_time: payload[:trigger_time],
+        description: settings[:description],
+        url: alert_link(payload[:alert][:id])
+      }
+      body[:description] = "[TEST] #{body[:description]}" if payload[:triggered_by_user_test]
+
+      # Property bag is a JSON blob that gets stored in Beacon along with the alert
+      body[:property_bag] = {
+        alert_id: "#{payload[:alert][:id]}",
+        alert_name: payload[:alert][:name],
+        alert_description: payload[:alert][:description].blank? ? payload[:alert][:name] : payload[:alert][:description]
+      }
+      unless payload[:alert][:runbook_url].blank?
+        body[:property_bag][:runbook_url] = payload[:alert][:runbook_url]
+      end
```

Comment:
No details on the violated conditions are sent?

Reply:
Outside of the url / id of the specific alert instance, is the id of the alert definition available to be sent as well?

Reply:
@mheffner thanks for the reminder, I'm working on adding violated conditions now.

@leecalcote the alert definition is included here https://github.com/appoptics/appoptics-services/pull/12/files/e343a77c7d197082379bc1f9e564ab57dae01558#diff-9542cb0142c71bf67c02763741be5b97R21 which should be unique.

Reply:
@jderrett got it. Good deal.